code/modules/power/multiz_cable.dm

```diff
@@ -0,0 +1,65 @@
+/obj/machinery/power/deck_relay //This bridges powernets
+	name = "Multi-deck power adapter"
+	desc = "This impressive machine uses plasma based power transmission technology to supply power to alternate decks. It requires a steady power source. Click it with a multitool / ODN scanner to see stats and reacquire connections."
+	icon = 'icons/obj/power.dmi'
+	icon_state = "cablerelay-off"
+	var/obj/machinery/power/deck_relay/below
+	var/obj/machinery/power/deck_relay/above
+	var/list/relays = list() //to bridge the powernets.
+	anchored = TRUE
+	density = FALSE
+
+/obj/machinery/power/deck_relay/process()
+	refresh() //Sometimes the powernets get lost, so we need to keep checking.
+	if(powernet && (powernet.avail <= 0))		// is it powered?
+		icon_state = "cablerelay-off"
+	else
+		icon_state = "cablerelay-on"
+	if(!below || QDELETED(below) || !above || QDELETED(above))
+		icon_state = "cablerelay-off"
+		find_relays()
```

Comment:
this seems rather excessive to run every tick of process if either above or below aren't present.

Reply:
this should not be a tick at all, just have a global listener for deck relay creation and notify all nearby relays at the time of creation/deletion